src/index.ts

```diff
@@ -0,0 +1,227 @@
+        const [freeSlots, bookedSlots] =
+            (await bookingService.getFreeSlotsForWeek(id, dateToCheck)) as any;
+
+        if (freeSlots.length > 0) {
+            if (!pendingBookingPromise) {
+                logger.success(
+                    `${name} (${getShortDate(dateToCheck)}) ${
+                        freeSlots.length
+                    } free time slots`
+                );
+                pendingBookingPromise = handleBooking(
+                    bookingService,
+                    freeSlots,
+                    id
+                );
+                await pendingBookingPromise;
+                pendingBookingPromise = undefined;
+            }
+        } else {
+            logger.verbose(
+                `${name} (${getShortDate(dateToCheck)}) 0 free - ${
+                    bookedSlots.length
+                } reserved`
+            );
+            dateToCheck = addDays(dateToCheck);
+        }
+
+        tracker.track();
+
+        if (config.throttle) {
+            await new Promise((resolve) =>
+                setTimeout(resolve, config.throttle * 1000)
+            );
+        }
+    }
+    logger.debug("Max date reached, checking next location...");
+
+    locationQueue.push(locationQueue.shift());
+    checkAvailableSlotsForLocation(
+        bookingService,
+        locationQueue,
+        getStartOfWeekDate(config.min_date || new Date())
+    );
+}
+
+async function handleBooking(
+    bookingService: BookingService,
+    freeSlots: any,
+    locationId: string
+) {
+    const { parent } = freeSlots.at(0);
+    const serviceTypeId = parent.attribs["data-servicetypeid"];
+    const timeslot = parent.attribs["data-fromdatetime"];
+
+    const booking = await bookingService.bookSlot(
+        serviceTypeId,
+        timeslot,
+        locationId,
+        config
+    );
+
+    if (booking) {
+        logger.success("BOOKING SUCCESSFUL");
+        logger.success(`Booking number: \t ${booking.bookingNumber}`);
+        logger.success(`Time: \t\t\t ${booking.slot}`);
+        logger.success(`Location: \t\t ${booking.expedition}`);
+        logger.success(`Email: \t\t\t ${config.email}`);
+        logger.success(`Phone: \t\t\t ${config.phone}`);
+        process.exit();
+    } else {
+        logger.error(`Failed booking slot ${timeslot}`);
+
+        if (await bookingService.recover()) {
+            logger.info("Recovered booking session");
+        } else {
+            logger.error("Could not recover booking session...");
+            process.exit();
+        }
+    }
+}
+
+function shuffleArray(array: any[]) {
+    return array
+        .map((value) => ({ value, sort: Math.random() }))
+        .sort((a, b) => a.sort - b.sort)
+        .map(({ value }) => value);
+}
+
+function randomDate(start: Date, end: Date) {
+    const date = new Date(
+        start.getTime() + Math.random() * (end.getTime() - start.getTime())
+    );
+
```

Comment:
Minor thing, but we don't need spacing between everything, makes it harder to read in the end.